ibis/pandas/aggcontext.py

```diff
@@ -405,17 +407,67 @@ def sliced_agg(s):
 
         # perform the per-group rolling operation
         windowed = self.construct_window(grouped)
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                "ignore", message=".+raw=True.+", category=FutureWarning
+
+        if callable(function):
+            # Use custom logic to computing rolling window UDF instead of
+            # using pandas's rolling function.
+            # This is because pandas's rolling function doesn't support
+            # multi param UDFs.
+            def create_input_gen(grouped_series, window_size):
+                # create a generator for each input series
+                # the generator will yield a slice of the
+                # input series for each valid window
+                data = getattr(grouped_series, 'obj', grouped_series).values
+                window_size_array = window_size.values
+                for i in range(len(window_size_array)):
+                    k = window_size.index[i]
+                    yield data[k - window_size_array[i] + 1 : k + 1]
+
+            # Compute window indices and manually roll
+            # over the window.
+            # If an window has only nan values, we output nan for
+            # the window result. This follows pandas rolling apply
+            # behavior.
+            raw_window_size = windowed.apply(len, raw=True).reset_index(
+                drop=True
+            )
+            mask = ~(raw_window_size.isna())
+            window_size = raw_window_size[mask].astype('i8')
+            window_size_array = window_size.values
+
+            # If there is no args, then the UDF only takes a single
+            # input which is defined by grouped_data
+            # This is a complication due to the lack of standard
+            # way to pass multiple input Series/SeriesGroupBy
+            # to AggregationContext.agg()
+            inputs = args if len(args) > 0 else [grouped_data]
+
+            input_gens = list(
+                create_input_gen(arg, window_size)
+                if isinstance(arg, (pd.Series, SeriesGroupBy))
+                else itertools.repeat(arg)
+                for arg in inputs
+            )
+
+            valid_result = pd.Series(
+                function(*(next(gen) for gen in input_gens))
+                for i in range(len(window_size_array))
             )
+
+            valid_result = pd.Series(valid_result)
+            valid_result.index = window_size.index
```

Comment:
are you testing the output indexes are correct?

Reply:
yes there are tests (test_udaf_window_interval) that cover out of order indices and make sure the output is correct